mirage/lE_http_server.ml

```diff
@@ -73,7 +73,7 @@ struct
       Alpn.injection;
     }
 
-  let with_lets_encrypt_certificates ?(port = 443) stackv4v6 ~production config
+  let with_lets_encrypt_certificates ?(port = 443) ?(alpn_protocols= [ "http/1.1"; "h2" ]) stackv4v6 ~production config
```

Comment:
The order was changed
```suggestion
  let with_lets_encrypt_certificates ?(port = 443) ?(alpn_protocols= [ "h2"; "http/1.1" ]) stackv4v6 ~production config
```

Reply:
Yes, I would like to keep `http/1.1` by default and when the user wants to handle `h2`, it must write explicitely.

Reply:
So indeed the order is relevant here: a client proposes a list of ALPN in its client hello, and the server selects one ALPN by using a "first match" from the ones configured for the server (in the case above `"http/1.1"; "h2"` -- previously `"h2"; "http/1.1"`) and the list received by the client.

So, TL;DR: should we prioritize HTTP1.1 or HTTP2? If "it seems that the `h2` server has some bugs", maybe it is better to only advertise http1 by default? In any case, we should more clearly document the default setting (including priority) :)

Reply:
I improved the documentation to explain the order of protocols and what is the default value.